components/statistics/portal_statistics.lua

```diff
@@ -0,0 +1,1413 @@
+		end
+		return opponentNames
+	end
+
+	local processData = function(item)
+		local placement = string.sub(item.placement, 1, 1)
+		for _, opponent in pairs(makeOpponentTable(item) or {}) do
+			if not data[opponent] then
+				data[opponent] = {['1'] = 0, ['2'] =  0, ['3'] = 0, showWins = 0, sWinData = {}}
+			end
+			if placement == FIRST and item.liquipediatier == TIER1 and item.liquipediatiertype ~= SHOWMATCH then
+				table.insert(data[opponent].sWinData, {
+						icon = item.icon,
+						iconDark = item.icondark,
+						pagename = item.pagename,
+						shortname = item.shortname
+					}
+				)
+			end
+			if placement == FIRST and item.liquipediatiertype == SHOWMATCH then
+				data[opponent].showWins = data[opponent].showWins + 1
+			elseif item.liquipediatiertype ~= SHOWMATCH then
+				data[opponent][placement] = data[opponent][placement] + 1
+			end
+		end
+	end
+
+	Lpdb.executeMassQuery('placement', queryParameters, processData)
+
+	return data
+end
+
+
+--[[
+Section: Display Functions
+]]--
+
+
+---@param args table
+---@param parameters table
+---@param tagType string
+---@return Html
+function StatisticsPortal._returnGameCell(args, parameters, tagType)
+	local gameCell = mw.html.create(tagType)
+	if Logic.readBool(args.multiGame) and not parameters.game then
+		gameCell:wikitext('Total')
+	elseif Logic.readBool(args.multiGame) then
+		gameCell:wikitext(parameters.game)
+	end
+	return gameCell
+end
+
+
+---@param args table
+---@return Html
+function StatisticsPortal._earningsTableHeader(args)
+	local columnText = args.opponentType == Opponent.team and 'Organization' or 'Player'
+
+	local row = mw.html.create('tr')
+		:tag('th'):wikitext('#'):addClass('unsortable'):done()
+		:tag('th'):wikitext(columnText):addClass('unsortable'):done()
+		:tag('th'):wikitext('Achievements'):css('width', '200px'):addClass('unsortable'):done()
+		:tag('th'):wikitext(Medal['1']):done()
+		:tag('th'):wikitext(Medal['2']):done()
+		:tag('th'):wikitext(Medal['3']):done()
+
+	if Logic.readBool(args.displayShowMatches) then
+		row:tag('th'):wikitext('Show<br>Match')
+	end
+
+	row:tag('th')
+		:tag('abbr')
+		:attr('title', 'Total earnings across all games')
+		:wikitext('Earnings')
+
+	return row
+end
+
+
+---@param args table
+---@param placements table
+---@param earnings number
+---@param opponentIndex number
+---@param opponentDisplay Html
+---@return Html
+function StatisticsPortal._earningsTableRow(args, placements, earnings, opponentIndex, opponentDisplay)
+	local row = mw.html.create('tr')
+		:css('line-height', '25px')
+		:css('text-align', 'center')
+		:tag('td'):wikitext(opponentIndex):done()
+		:tag('td'):css('text-align', 'left'):node(opponentDisplay):done()
+		:tag('td'):wikitext(StatisticsPortal._achievementsDisplay(placements.sWinData or {})):done()
+		:tag('td'):wikitext(placements['1'] or '0'):done()
+		:tag('td'):wikitext(placements['2'] or '0'):done()
+		:tag('td'):wikitext(placements['3'] or '0'):done()
```

Comment:
Will this consider `3-4` placements as 3rd? They are very commons (Single Elimination no 3rd place match)

Reply:
By default yes, The allowed placements are set as a constant local variable DEFUALT_ALLOWED_PLACES on line 42. These can be overwritten on a per module call basis based on a user defined string of desired placements to include in the tally (Line 663).

Reply:
Where is it `3-4` merged into `3`?

Reply:
Line 981, it takes the placement value and extracts the first character of the string